src/network/networkd-conf.c

```diff
@@ -180,3 +182,49 @@ int config_parse_duid_rawdata(
         ret->raw_data_len = count;
         return 0;
 }
+
+static const char* const ip_service_type_table[_IP_SERVICE_TYPE_MAX] = {
+        [IP_SERVICE_TYPE_CS4] = "CS4",
+        [IP_SERVICE_TYPE_CS6] = "CS6",
+};
+DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING(ip_service_type, IPServiceType);
+
+int config_parse_ip_service_type(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        IPServiceType type;
+        int *ip_service_type = data;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        type = ip_service_type_from_string(rvalue);
```

Comment:
Using `_from_string()` looks overkill and redundant here. How about the following
```
if (streq(rvalue, "CS4"))
    *ip_service_type = IPTOS_CLASS_CS4;
else if (streq(rvalue, "CS6"))
    *ip_service_type = IPTOS_CLASS_CS6;
else {
    log_syntax(...);
    return 0;
}
```

Reply:
Done.